src/ui/ui_settings/__tests__/ui_settings_service.js

```diff
@@ -0,0 +1,410 @@
+import { isEqual } from 'lodash';
+import expect from 'expect.js';
+import { errors as esErrors } from 'elasticsearch';
+
+import { getDefaultSettings } from '../defaults';
+import { UiSettingsService } from '../ui_settings_service';
+
+import {
+  assertPromise,
+  assertRejection,
+  createCallClusterStub,
+} from './lib';
+
+const INDEX = '.kibana';
+const TYPE = 'config';
+const ID = 'kibana-version';
+
+function setup(options = {}) {
+  const {
+    readInterceptor,
+    esDocSource = {},
+    callCluster = createCallClusterStub(INDEX, TYPE, ID, esDocSource)
+  } = options;
+
+  const uiSettings = new UiSettingsService({
+    index: INDEX,
+    type: TYPE,
+    id: ID,
+    readInterceptor,
+    callCluster,
+  });
+
+  return {
+    uiSettings,
+    assertGetQuery: callCluster.assertGetQuery,
+    assertUpdateQuery: callCluster.assertUpdateQuery,
+  };
+}
+
+describe('ui settings', function () {
+  describe('overview', function () {
+    it('has expected api surface', function () {
+      const { uiSettings } = setup();
+      expect(typeof uiSettings.get).to.equal('function');
```

Comment:
Hold over from the previous tests, but happy to fix and get better error messages.